drivers/serial/uart_mchp_xec.c

```diff
@@ -749,8 +749,10 @@ static int uart_xec_irq_tx_complete(const struct device *dev)
 	struct uart_regs *regs = dev_cfg->regs;
 	k_spinlock_key_t key = k_spin_lock(&dev_data->lock);
 
-	int ret = ((regs->LSR & (LSR_TEMT | LSR_THRE))
-				== (LSR_TEMT | LSR_THRE)) ? 1 : 0;
+	int ret = (((regs->IER & IER_TBE) == 0) &&
```

Comment:
Nit - this logic is getting complicated for a ternary operator.  Can you convert to an if/else?